user/src/com/google/gwt/user/server/rpc/RemoteServiceServlet.java

```diff
@@ -94,6 +96,20 @@ static SerializationPolicy loadSerializationPolicy(HttpServlet servlet,
           try {
             serializationPolicy = SerializationPolicyLoader.loadFromStream(is,
                 null);
+            if (serializationPolicy.hasClientFields()) {
+              if (ENABLE_ENHANCED_CLASSES) {
+                servlet.log("WARNING: Service deserializes enhanced JPA/JDO classes, which is " +
+                        "unsafe. See https://github.com/gwtproject/gwt/issues/9709 for more " +
+                        "detail on the vulnerability that this presents.");
```

Comment:
it should be: "details on the vulnerability"

Reply:
As German I tend to use `details` because that is how we speak natively but English also has "more detail on" if you want to emphasize that some content exists which provides a more granular version of what has already been said. So I think `for more detail on` is correct here as the previous sentence has given a very condensed summary. I think "more details on" would be more appropriate if the text has already given some details but not all.
But as I am not a native speaker I am fine with whatever @niloc132 choses :)

Reply:
yeah, as German I would say 'details'. Checked also on Google both is ok. It is up to @niloc132 :-)

Reply:
Both "sound" correct to me - it appears that "for more detail" would be incorrect, but "for more details" or "for more detail on" are both acceptable? I'm going to leave this as is.